Sming/Components/mdns/include/Network/Mdns.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <user_config.h>
+#include <Data/CStringArray.h>
+#include <memory>
+
+#ifdef LWIP_MDNS_RESPONDER
+#include <lwip/apps/mdns.h>
+#endif
+
+namespace mDNS
+{
+struct Service {
+	enum class Protocol {
+		Udp,
+		Tcp,
+	};
+
+	const char* name;
+	const char* type;
+	Protocol protocol;
+	uint16_t port;
+};
+
+class Responder
+{
+public:
+	/**
+	 * @brief Obtain text chunks for a service via callback
+	 *
+	 * LWIP2 calls this each time a TXT reply is created
+	 * Other implementations call it via begin().
+	 *
+	 * @param svc The service description
+	 * @retval CStringArray List of name=value pairs
+	 */
+	using ServiceTextDelegate = Delegate<CStringArray(const Service& svc)>;
+
+	/**
+	 * @brief Initialise the responder
+	 * @param hostname
+	 * @param getServiceText Callback invoked for a service to obtain TXT information
+	 * @retval bool true on success
+	 */
+	bool init(const String& hostname, ServiceTextDelegate getServiceText);
```

Comment:
Could the `getServiceText` argument be moved to `addService` or a separate method?
I guess "hostname.local" discovery will be sufficient for many people, so let's not confuse them with service records.

Reply:
Agreed. I think what's confusing me is that the old `mdns_info` interface for LWIP1 mixes up host discovery with the PUCK service. I had to dig around to find out what that was!